### PR TITLE
6u - implemented connection to db

### PR DESCRIPTION
**Pull request by MaksKulykov**

### CODE REVIEW

File: src/maks.kulykov/DatabaseManager.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class DatabaseManager {
+    private String db = "mydb";
+    private String userName = "postgres";
+    private String password = "postgres";
+
+    private String cmdDB;
```
Comment (OlgaKulikova) [inline src/maks.kulykov/DatabaseManager.java]: I think you should be consistent in the naming style - camelCase. So I think it is better 'cmdDb'.

File: src/maks.kulykov/DatabaseManager.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    private String cmdPassword;
+
+    public DatabaseManager(String db, String userName, String password) {
+        this.cmdDB = db;
+        this.cmdUserName = userName;
+        this.cmdPassword = password;
+    }
+
+    public String checkCredentials() {
+        String resp = "";
```
Comment (OlgaKulikova) [inline src/maks.kulykov/DatabaseManager.java]: this short name is not clear, don't shy  use the full words

Comment (OlgaKulikova) [reply src/maks.kulykov/DatabaseManager.java]: and if I don't know what do you planned to implement I won't guess what is resp means. Naming should help to understand the logic. I'd name it like 'wrongCredential'

File: src/maks.kulykov/DatabaseManager.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+public class DatabaseManager {
+    private String db = "mydb";
+    private String userName = "postgres";
+    private String password = "postgres";
+
+    private String cmdDB;
+    private String cmdUserName;
+    private String cmdPassword;
+
+    public DatabaseManager(String db, String userName, String password) {
```
Comment (OlgaKulikova) [inline src/maks.kulykov/DatabaseManager.java]: I'd add user's input as parameters in 'checkCredentials()'. Then you don't need to save them as class fields.

File: src/maks.kulykov/DatabaseManager.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            resp = "username";
+        } else if (!cmdPassword.equals(password)) {
+            resp = "password";
+        }
+        return resp;
+    }
+
+    public void connection() {
+        try (Connection conn = DriverManager.getConnection(
+                "jdbc:postgresql://127.0.0.1:5432/" + cmdDB, cmdUserName, cmdPassword)) {
```
Comment (OlgaKulikova) [inline src/maks.kulykov/DatabaseManager.java]: It's a good practice to save literals to the constants.

Comment (OlgaKulikova) [reply src/maks.kulykov/DatabaseManager.java]: and here you can use your constant values of db name, user name and password. If you already checked user input before.

File: src/maks.kulykov/DatabaseManager.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            if (conn != null) {
+                System.out.println("Connected to the database!");
```
Comment (OlgaKulikova) [inline src/maks.kulykov/DatabaseManager.java]: Better to return the string and make 'sout' in the main class.

File: src/maks.kulykov/Main.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        DatabaseManager manager = new DatabaseManager(dbName, userName, password);
+
+        boolean isConnection = false;
+
+        while (!isConnection) {
+            switch (manager.checkCredentials()) {
+                case "db" -> {
+                    System.out.println("There is no database with this name. Please enter correct database:");
+                    dbName = reader.readLine();
+                    manager = new DatabaseManager(dbName, userName, password);
```
Comment (OlgaKulikova) [inline src/maks.kulykov/Main.java]: it is also weird to create a DatabaseManager again. So need to rewrite this logic with passing credentials as parameters to 'checkCredentials()'.
Also think about to call method not in every case but once after switch().

File: test/maks/kulykov/MainTest.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
+package maks.kulykov;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class MainTest {
+
+    @Test
+    public void testWrongDBName() {
```
Comment (OlgaKulikova) [inline test/maks/kulykov/MainTest.java]: also 'DB' as camel case - > 'Db'

File: src/maks.kulykov/DatabaseManager.java (hunk 1 above)
Comment (MaksKulykov) [reply src/maks.kulykov/DatabaseManager.java]: +

File: src/maks.kulykov/DatabaseManager.java (hunk 2 above)
Comment (MaksKulykov) [reply src/maks.kulykov/DatabaseManager.java]: +

File: src/maks.kulykov/DatabaseManager.java (hunk 3 above)
Comment (MaksKulykov) [reply src/maks.kulykov/DatabaseManager.java]: +

File: src/maks.kulykov/DatabaseManager.java (hunk 4 above)
Comment (MaksKulykov) [reply src/maks.kulykov/DatabaseManager.java]: +

File: src/maks.kulykov/Main.java (hunk 6 above)
Comment (MaksKulykov) [reply src/maks.kulykov/Main.java]: +

File: test/maks/kulykov/MainTest.java (hunk 7 above)
Comment (MaksKulykov) [reply test/maks/kulykov/MainTest.java]: +

File: src/maks.kulykov/DatabaseManager.java (hunk 5 above)
Comment (MaksKulykov) [reply src/maks.kulykov/DatabaseManager.java]: +

**Pull request merged**